### ISSUE TITLE
bug leading to recursive list

### ISSUE THREAD

**Issue by sdementen** [cdgriffith/Box#68]
The following code should print `[['foo']]` but instead prints `[[...]]` showing that a auto-referential list has been created.
```python
from box import Box

def bl():
    b = Box()
    bl = b.setdefault("l", [])
    bl.append(["foo"])
    print(bl)
```

Issue lies on https://github.com/cdgriffith/Box/blob/master/box.py#L771 (or in the append function at https://github.com/cdgriffith/Box/blob/master/box.py#L802) that detects self referential entities.
The `id(iterable)` is not robust enought as two object may have the same id if the first object has been garbage collected.
One can see this in the example below
```python
def bl():
    b = Box()
    l1 = []
    bl = b.setdefault("l", l1)
    del l1    # comment this to see that now it works as l1 is not garbage collected
    bl.append(["foo"])
    print(bl)
```
A fix could be to replace https://github.com/cdgriffith/Box/blob/master/box.py#L771 by
```python
        self.box_org_ref = id(iterable) if iterable else 0
```

**Comment by cdgriffith** [comment]
@sdementen  Thank you for the bug report and your hard work figuring out the cause of the issue and it's solution. 

My apologizes for not noticing this earlier due to Holidays. I will work to get this fix in by this weekend!

**Comment by cdgriffith** [comment]
Fixed in 3.2.4 #72
